lib/python/qmk/cli/compile.py

```diff
@@ -28,6 +29,24 @@ def compile(cli):
     If --keyboard and --keymap are provided this command will build a firmware based on that.
 
     """
+    # Set CWD as directory command was issued from
+    cwd = os.path.normpath(os.path.join(os.environ['ORIG_CWD']))
+
+    # Initialize boolean to check for being in a keyboard directory and initialize keyboard string
+    in_keyboard = False
+    keyboard = ""
+
+    # Loop through three levels up to check for keyboards root directory
```

Comment:
I would probably approach this differently, something like:

```
keyboards_path = os.path.join(os.getcwd(), 'keyboards')
if os.environ['ORIG_CWD'].startswith(keyboards_path):
    localpath = os.environ['ORIG_CWD'][len(keyboards_path):]
    # Parse down from localpath here
```

Reply:
Ah, I like this. I will test this tonight!

Reply:
I just tested this, the main thing is that it doesn't fail quite as gracefully when running `qmk compile` in the `keyboards` directory. Maybe not a necessary feature, but I will leave that up to you